src/Symfony/Component/Messenger/Adapter/AmqpExt/AmqpFactory.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Adapter\AmqpExt;
+
+class AmqpFactory
+{
+    public function createConnection(array $credentials): \AMQPConnection
+    {
+        return new \AMQPConnection($credentials);
+    }
+
+    public function createChannel(\AMQPConnection $connection): \AMQPChannel
+    {
+        return new \AMQPChannel($connection);
+    }
+
+    public function createQueue(\AMQPChannel $channel): \AMQPQueue
+    {
+        return new \AMQPQueue($channel);
```

Comment:
I think these `\AMQP` classes are only in the `amqp` pecl extension? if so, it should be a requirement in `composer.json`
https://getcomposer.org/doc/04-schema.md#package-links
something like
```
"ext-amqp": "*"
```
unless there's an alternative?

Reply:
That's correct. We can't add the requirement in this `composer.json` file as it's not a requirement for the entire component, but for its AMQP adapter only. Though you are right in pointing that we don't handle properly the case when the extension is not installed: we can add a nice exception message when trying to create the adapter while the extension does not exist.

Reply:
good to know! I'm not going to hold up the train for that, but probably a good idea to add at some point.